app/services/marketplace_service/transaction.rb

```diff
@@ -29,6 +29,7 @@ module Entity
         :created_at,
         :availability,
         :booking_uuid,
+        :auth_data,
```

Comment:
The `Transaction` entity is used to convert Model objects to Hashes. See line: https://github.com/sharetribe/sharetribe/pull/2639/files#diff-e2097c0e3cb9d60a2686ec38668efb43L92

```
Transaction[EntityUtils.model_to_hash(transaction_model)]
```

It's good to keep in mind that the `Transaction` entity here is a little bit different that the other Entities you may find from the code base. As you can see, the entity is builded by using method `.define_entity`. Which is actually a bit older and simpler version of what we use today, `.define_builder`.

In the line above, we have the `transaction_model`, which is an ActiveRecord model. EntityUtils.model_to_hash takes the model `attributes` hash and symbolizes the keys. Then that hash is passed to `Transaction` entity. Entities are usually called with `.call`, but since the `.define_entity` method returns a lambda, and lambda aliases `.call` to `[]`, `Transaction[some_hash]` equals `Transaction.call(some_hash)`. This might be a bit misleading.

So the reason why `auth_data` is empty is that the `auth_data` can't be found from the model. I guess you should build the `auth_data` in the `paid` method where you are calling the `HarmonyClient`.